pkg/state/state.go

```diff
@@ -2192,9 +2192,14 @@ func ConditionEnabled(r ReleaseSpec, values map[string]interface{}) (bool, error
 	if len(r.Condition) == 0 {
 		return true, nil
 	}
+
+	if conditionMatch, err := strconv.ParseBool(r.Condition); err == nil {
+		return conditionMatch, nil
+	}
+
 	conditionSplit := strings.Split(r.Condition, ".")
 	if len(conditionSplit) != 2 {
-		return false, fmt.Errorf("Condition value must be in the form 'foo.enabled' where 'foo' can be modified as necessary")
+		return false, fmt.Errorf("condition value must be either in the form 'foo.enabled' where 'foo' can be modified as necessary or string that can be evaluated to bool")
```

Comment:
I contend that introducing unnecessary complexity serves no purpose. The issue at hand aptly demonstrates this point. My reasoning is influenced by Helm's elegant handling of its `dependencies[].condition` parameter in the `Chart.yaml` file. This parameter singularly points to another key in `$.Values`, without any concern for the data type—be it boolean, string, or map. To elaborate, I often employ a key named `enabled`, which can take on a variety of values—`true`, `"true"`, or more specialized values like `"manual"` or `"auto"`. Such values are evaluated as `true` for dependency conditions for the simple reason that they are not null. Any deviation from this approach would be excessive. The condition is met when the targeted key is present and its value is not empty; otherwise, the condition remains unfulfilled. The choice of how to manage this rests solely with the Helm chart developer.

In the same vein, I propose the following criteria for this case, which negates the need for type-specific considerations:

- **Existence of Key**
  - Contains a non-empty value? ➜ **Condition: True**
  - Lacks value? ➜ **Condition: False**
- **Absence of Key** ➜ **Condition: False**